packages/protocol/src/protocol/message_layer/StreamMessage.ts

```diff
@@ -122,15 +116,12 @@ export default class StreamMessage<T = unknown> {
         validateIsType('signature', signature, 'Uint8Array', Uint8Array)
         this.signature = signature
 
-        if (typeof content === 'string') {
-            // this.parsedContent gets written lazily
-            this.serializedContent = content
-        } else {
-            this.parsedContent = content
-            this.serializedContent = JSON.stringify(content)
-        }
+        StreamMessage.validateContentType(contentType)
+        this.contentType = contentType
 
-        validateIsNotEmptyString('content', this.serializedContent)
+        this.serializedContent = content
+
+        validateIsNotEmptyByteArray('content', this.serializedContent)
```

Comment:
Why empty is not ok?

Reply:
The content has to be a valid JSON as binary content type is not supported yet in this PR.